test/nodetool/test_status.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+
+from collections import defaultdict
+from enum import Enum
+from rest_api_mock import expected_request
+from socket import getnameinfo
+from typing import NamedTuple
+
+import pytest
+
+
+class NodeStatus(Enum):
+    Up = 'U'
+    Down = 'D'
+    Unknown = '?'
+
+
+class NodeState(Enum):
+    Joining = 'J'
+    Leaving = 'L'
+    Moving = 'M'
+    Normal = 'N'
+
+
+class Node(NamedTuple):
+    endpoint: str
+    host_id: str
+    load: int
+    tokens: list[str]
+    datacenter: str
+    rack: str
+    status: NodeStatus
+    state: NodeState
+
+
+null_ownership_error = ("Non-system keyspaces don't have the same replication settings, "
+                        "effective ownership information is meaningless")
+
+
+def validate_status_output(res, keyspace, nodes, ownership, resolve):
+    datacenters = sorted(list(set([node.datacenter for node in nodes.values()])))
+    load_multiplier = {"KB": 1024, "MB": 1024**2, "GB": 1024**3, "TB": 1024**4}
+
+    lines = res.split('\n')
+    i = 0
+
+    for dc in datacenters:
+        dc_line = lines[i]
+        assert dc_line.startswith("Datacenter:")
+        res_dc = dc_line.split()[1]
+        assert dc == res_dc
+        dc_line_len = len(dc_line)
+
+        i += 1
+        assert lines[i] == "=" * dc_line_len
+
+        i += 1
+        assert lines[i] == "Status=Up/Down"
+
+        i += 1
+        assert lines[i] == "|/ State=Normal/Leaving/Joining/Moving"
+
+        i += 1
+        if keyspace is None:
+            assert lines[i].split() == ["--", "Address", "Load", "Tokens", "Owns", "Host", "ID", "Rack"]
+        else:
+            assert lines[i].split() == ["--", "Address", "Load", "Tokens", "Owns", "(effective)", "Host", "ID", "Rack"]
+
+        dc_eps = {ep for ep, node in nodes.items() if node.datacenter == dc}
+        if resolve:
+            name_to_ep = {getnameinfo((ep, 0), 0)[0]: ep for ep in nodes.keys()}
```

Comment:
this reminds me that i need to improve `test/nodetool/test_ring.py` to test the `resolve-ip` command line option.

Reply:
created #17553 to address this.

Reply:
Thanks.